{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "spades = '/Users/olgabot/anaconda3/envs/winky/bin/spades.py'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SPAdes genome assembler v3.11.0\r\n",
      "\r\n",
      "Usage: /Users/olgabot/anaconda3/envs/winky/bin/spades.py [options] -o <output_dir>\r\n",
      "\r\n",
      "Basic options:\r\n",
      "-o\t<output_dir>\tdirectory to store all the resulting files (required)\r\n",
      "--sc\t\t\tthis flag is required for MDA (single-cell) data\r\n",
      "--meta\t\t\tthis flag is required for metagenomic sample data\r\n",
      "--rna\t\t\tthis flag is required for RNA-Seq data \r\n",
      "--plasmid\t\truns plasmidSPAdes pipeline for plasmid detection \r\n",
      "--iontorrent\t\tthis flag is required for IonTorrent data\r\n",
      "--test\t\t\truns SPAdes on toy dataset\r\n",
      "-h/--help\t\tprints this usage message\r\n",
      "-v/--version\t\tprints version\r\n",
      "\r\n",
      "Input data:\r\n",
      "--12\t<filename>\tfile with interlaced forward and reverse paired-end reads\r\n",
      "-1\t<filename>\tfile with forward paired-end reads\r\n",
      "-2\t<filename>\tfile with reverse paired-end reads\r\n",
      "-s\t<filename>\tfile with unpaired reads\r\n",
      "--pe<#>-12\t<filename>\tfile with interlaced reads for paired-end library number <#> (<#> = 1,2,..,9)\r\n",
      "--pe<#>-1\t<filename>\tfile with forward reads for paired-end library number <#> (<#> = 1,2,..,9)\r\n",
      "--pe<#>-2\t<filename>\tfile with reverse reads for paired-end library number <#> (<#> = 1,2,..,9)\r\n",
      "--pe<#>-s\t<filename>\tfile with unpaired reads for paired-end library number <#> (<#> = 1,2,..,9)\r\n",
      "--pe<#>-<or>\torientation of reads for paired-end library number <#> (<#> = 1,2,..,9; <or> = fr, rf, ff)\r\n",
      "--s<#>\t\t<filename>\tfile with unpaired reads for single reads library number <#> (<#> = 1,2,..,9)\r\n",
      "--mp<#>-12\t<filename>\tfile with interlaced reads for mate-pair library number <#> (<#> = 1,2,..,9)\r\n",
      "--mp<#>-1\t<filename>\tfile with forward reads for mate-pair library number <#> (<#> = 1,2,..,9)\r\n",
      "--mp<#>-2\t<filename>\tfile with reverse reads for mate-pair library number <#> (<#> = 1,2,..,9)\r\n",
      "--mp<#>-s\t<filename>\tfile with unpaired reads for mate-pair library number <#> (<#> = 1,2,..,9)\r\n",
      "--mp<#>-<or>\torientation of reads for mate-pair library number <#> (<#> = 1,2,..,9; <or> = fr, rf, ff)\r\n",
      "--hqmp<#>-12\t<filename>\tfile with interlaced reads for high-quality mate-pair library number <#> (<#> = 1,2,..,9)\r\n",
      "--hqmp<#>-1\t<filename>\tfile with forward reads for high-quality mate-pair library number <#> (<#> = 1,2,..,9)\r\n",
      "--hqmp<#>-2\t<filename>\tfile with reverse reads for high-quality mate-pair library number <#> (<#> = 1,2,..,9)\r\n",
      "--hqmp<#>-s\t<filename>\tfile with unpaired reads for high-quality mate-pair library number <#> (<#> = 1,2,..,9)\r\n",
      "--hqmp<#>-<or>\torientation of reads for high-quality mate-pair library number <#> (<#> = 1,2,..,9; <or> = fr, rf, ff)\r\n",
      "--nxmate<#>-1\t<filename>\tfile with forward reads for Lucigen NxMate library number <#> (<#> = 1,2,..,9)\r\n",
      "--nxmate<#>-2\t<filename>\tfile with reverse reads for Lucigen NxMate library number <#> (<#> = 1,2,..,9)\r\n",
      "--sanger\t<filename>\tfile with Sanger reads\r\n",
      "--pacbio\t<filename>\tfile with PacBio reads\r\n",
      "--nanopore\t<filename>\tfile with Nanopore reads\r\n",
      "--tslr\t<filename>\tfile with TSLR-contigs\r\n",
      "--trusted-contigs\t<filename>\tfile with trusted contigs\r\n",
      "--untrusted-contigs\t<filename>\tfile with untrusted contigs\r\n",
      "\r\n",
      "Pipeline options:\r\n",
      "--only-error-correction\truns only read error correction (without assembling)\r\n",
      "--only-assembler\truns only assembling (without read error correction)\r\n",
      "--careful\t\ttries to reduce number of mismatches and short indels\r\n",
      "--continue\t\tcontinue run from the last available check-point\r\n",
      "--restart-from\t<cp>\trestart run with updated options and from the specified check-point ('ec', 'as', 'k<int>', 'mc')\r\n",
      "--disable-gzip-output\tforces error correction not to compress the corrected reads\r\n",
      "--disable-rr\t\tdisables repeat resolution stage of assembling\r\n",
      "\r\n",
      "Advanced options:\r\n",
      "--dataset\t<filename>\tfile with dataset description in YAML format\r\n",
      "-t/--threads\t<int>\t\tnumber of threads\r\n",
      "\t\t\t\t[default: 16]\r\n",
      "-m/--memory\t<int>\t\tRAM limit for SPAdes in Gb (terminates if exceeded)\r\n",
      "\t\t\t\t[default: 250]\r\n",
      "--tmp-dir\t<dirname>\tdirectory for temporary files\r\n",
      "\t\t\t\t[default: <output_dir>/tmp]\r\n",
      "-k\t\t<int,int,...>\tcomma-separated list of k-mer sizes (must be odd and\r\n",
      "\t\t\t\tless than 128) [default: 'auto']\r\n",
      "--cov-cutoff\t<float>\t\tcoverage cutoff value (a positive float number, or 'auto', or 'off') [default: 'off']\r\n",
      "--phred-offset\t<33 or 64>\tPHRED quality offset in the input reads (33 or 64)\r\n",
      "\t\t\t\t[default: auto-detect]\r\n"
     ]
    }
   ],
   "source": [
    "! $spades"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "== Error ==  file with reads has unsupported format (only .fq, .fastq, .bam, .fa, .fasta, .fq.gz, .fastq.gz, .bam.gz, .fa.gz, .fasta.gz are supported): /Users/olgabot/code/winky/notebooks/ZN-1_ZNif1__2017-06-07_B01.ab1 (single reads, library number: 1, library type: sanger)\r\n",
      "\r\n",
      "In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru\r\n",
      "or report an issue on our GitHub repository github.com/ablab/spades\r\n",
      "Please provide us with params.txt and spades.log files from the output directory.\r\n"
     ]
    }
   ],
   "source": [
    "! $spades --sanger ZN-1_ZNif1__2017-06-07_B01.ab1 --sanger ZN-1_ZNir1__2017-06-07_E01.ab1 -o spades-ZN-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command line: /Users/olgabot/anaconda3/envs/winky/bin/spades.py\t--12\t/Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq\t-o\t/Users/olgabot/code/winky/notebooks/spades-ZN-1\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.11.0\n",
      "  Python version: 3.5.2\n",
      "  OS: Darwin-16.7.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olgabot/code/winky/notebooks/spades-ZN-1\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: not specified\n",
      "      right reads: not specified\n",
      "      interlaced reads: ['/Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq']\n",
      "      single reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed (with gzip)\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olgabot/code/winky/notebooks/spades-ZN-1/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olgabot/code/winky/notebooks/spades-ZN-1/spades.log\n",
      "\n",
      "== Splitting /Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq into left and right reads (in /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input directory)\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /Users/olgabot/anaconda3/envs/winky/share/spades-3.11.0-0/bin/hammer /Users/olgabot/code/winky/notebooks/spades-ZN-1/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  83)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.hpp          :  51)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  93)   Trying to determine PHRED offset\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  :  99)   Determined value is 33\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  : 120)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.002     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.002   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_1.fastq\n",
      "  0:00:00.010   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.012   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 111)   Total 4 reads processed\n",
      "  0:00:00.012     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.022     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 5214 kmers in total.\n",
      "  0:00:00.022     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.026     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.054     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.056     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 11512 bytes occupied (17.6632 bits per kmer).\n",
      "  0:00:00.056     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 357)   Arranging kmers in hash map order\n",
      "  0:00:00.056     4M / 616M  INFO    General                 (main.cpp                  : 155)   Clustering Hamming graph.\n",
      "  0:00:00.072     4M / 616M  INFO    General                 (main.cpp                  : 162)   Extracting clusters\n",
      "  0:00:00.074     4M / 616M  INFO    General                 (main.cpp                  : 174)   Clustering done. Total clusters: 5194\n",
      "  0:00:00.074     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 379)   Collecting K-mer information, this takes a while.\n",
      "  0:00:00.074     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_1.fastq\n",
      "  0:00:00.075     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.076     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 392)   Collection done, postprocessing.\n",
      "  0:00:00.076     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 406)   There are 5214 kmers in total. Among them 3374 (64.7104%) are singletons.\n",
      "  0:00:00.076     4M / 616M  INFO    General                 (main.cpp                  : 180)   Subclustering Hamming graph\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 5174. Among them 4208 (81.3297%) are good\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 0. Among them 0 (nan%) are good\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 20. Among them 0 (0%) are good\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2 kmers\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 4208\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.971698,0.0188679,0.00943396,0),(0,0.932692,0.0384615,0.0288462),(0.0283019,0.0377358,0.915094,0.0188679),(0,0.00961538,0,0.990385))\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (main.cpp                  : 185)   Finished clustering.\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (main.cpp                  : 204)   Starting solid k-mers expansion in 1 threads.\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (main.cpp                  : 225)   Solid k-mers iteration 0 produced 0 new k-mers.\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (main.cpp                  : 229)   Solid k-mers finalized\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (hammer_tools.cpp          : 211)   Starting read correction in 1 threads.\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (hammer_tools.cpp          : 222)   Correcting pair of reads: /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_1.fastq and /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.087    28M / 616M  INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 0 of 2 reads.\n",
      "  0:00:00.088    28M / 616M  INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:00.088    28M / 616M  INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:00.092     4M / 616M  INFO    General                 (hammer_tools.cpp          : 270)   Correction done. Changed 0 bases in 0 reads.\n",
      "  0:00:00.092     4M / 616M  INFO    General                 (hammer_tools.cpp          : 271)   Failed to correct 0 bases out of 3824.\n",
      "  0:00:00.092     4M / 616M  INFO    General                 (main.cpp                  : 255)   Too few reads have changed in this iteration. Exiting.\n",
      "  0:00:00.092     4M / 616M  INFO    General                 (main.cpp                  : 262)   Saving corrected dataset description to /Users/olgabot/code/winky/notebooks/spades-ZN-1/corrected/corrected.yaml\n",
      "  0:00:00.093     4M / 616M  INFO    General                 (main.cpp                  : 269)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "== Dataset description file was created: /Users/olgabot/code/winky/notebooks/spades-ZN-1/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.hpp          :  51)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  91)   Assembling dataset (/Users/olgabot/code/winky/notebooks/spades-ZN-1/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  61)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  84)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  85)   Converting paired reads\n",
      "  0:00:00.122    68M / 132M  INFO    General                 (binary_converter.hpp      : 159)   2 reads written\n",
      "  0:00:00.127     4M / 132M  INFO    General                 (read_converter.hpp        :  94)   Converting single reads\n",
      "  0:00:00.360   132M / 260M  INFO    General                 (binary_converter.hpp      : 159)   0 reads written\n",
      "  0:00:00.375     4M / 260M  INFO    General                 (graph_construction.hpp    : 120)   Constructing DeBruijn graph for k=21\n",
      "  0:00:00.376     4M / 260M  INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 1 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.376     4M / 260M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.377     4M / 260M  INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.377     4M / 260M  INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 67108864\n",
      "  0:00:00.379   568M / 1G    INFO    General                 (kmer_splitters.hpp        : 297)   Adding contigs from previous K\n",
      "  0:00:00.379     4M / 1G    INFO    General                 (kmer_splitters.hpp        : 310)   Used 8 reads. Maximum read length 985\n",
      "  0:00:00.379     4M / 1G    INFO    General                 (kmer_splitters.hpp        : 311)   Average read length 956\n",
      "  0:00:00.379     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.381     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2607 kmers in total.\n",
      "  0:00:00.381     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.381     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.381     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.381     4M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.381     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.381     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.386   580M / 1G    INFO    General                 (kmer_splitters.hpp        : 381)   Processed 2607 kmers\n",
      "  0:00:00.386   580M / 1G    INFO    General                 (kmer_splitters.hpp        : 386)   Used 2607 kmers.\n",
      "  0:00:00.387     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.394     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2607 kmers in total.\n",
      "  0:00:00.394     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.397     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.421     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.423     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 10368 bytes occupied (31.8159 bits per kmer).\n",
      "  0:00:00.423     4M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  87)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.424     4M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  91)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.424     4M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:00.426     4M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   59 22-mers were removed by early tip clipper\n",
      "  0:00:00.426     4M / 1G    INFO    General                 (graph_construction.hpp    : 136)   Condensing graph\n",
      "  0:00:00.426     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:00.428     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 4 sequences extracted\n",
      "  0:00:00.429     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:00.429     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.429     4M / 1G    INFO    General                 (graph_construction.hpp    : 141)   Building index with from graph\n",
      "  0:00:00.430     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.430     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.430     4M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.430     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.430     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.434   580M / 1G    INFO    General                 (edge_index_builders.hpp   :  80)   Used 8 sequences.\n",
      "  0:00:00.434     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.442     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2548 kmers in total.\n",
      "  0:00:00.442     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.445     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.467     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.469     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 10352 bytes occupied (32.5024 bits per kmer).\n",
      "  0:00:00.469     4M / 1G    INFO    General                 (edge_index_builders.hpp   : 105)   Collecting k-mer coverage information from graph, this takes a while.\n",
      "  0:00:00.469     4M / 1G    INFO    General                 (edge_index.hpp            :  91)   Index refilled\n",
      "  0:00:00.470     4M / 1G    INFO    General                 (graph_construction.hpp    : 173)   Filling coverage index\n",
      "  0:00:00.470     4M / 1G    INFO    General                 (edge_index_builders.hpp   : 183)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:00.470     4M / 1G    INFO    General                 (graph_construction.hpp    : 175)   Filling coverage and flanking coverage from index\n",
      "  0:00:00.470     4M / 1G    INFO    General                 (construction.cpp          :  30)   Figured out: read length = 985\n",
      "  0:00:00.471     4M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == EC Threshold Finding\n",
      "=== Stack Trace ===\n",
      "0   spades                              0x0000000107d4ed6d _ZN5utils16print_stacktraceEv + 173\n",
      "1   spades                              0x0000000107fb6298 _ZN14coverage_model17KMerCoverageModel3FitEv + 328\n",
      "2   spades                              0x0000000107f7d48b _ZN14debruijn_graph17GenomicInfoFiller3runERNS_10graph_packINS_13DeBruijnGraphEEEPKc + 1947\n",
      "3   spades                              0x0000000107f49ff3 _ZN6spades12StageManager3runERN14debruijn_graph10graph_packINS1_13DeBruijnGraphEEEPKc + 931\n",
      "4   spades                              0x0000000107d4c0ab _ZN6spades15assemble_genomeEv + 7531\n",
      "5   spades                              0x0000000107d4dbbe main + 2894\n",
      "6   libdyld.dylib                       0x00007fff9591c235 start + 1\n",
      "7   ???                                 0x0000000000000002 0x0 + 2\n",
      "Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.\n",
      "Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.\n",
      "Assertion failed: (cov_.size() > 10), function Fit, file /anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp, line 187.\n",
      "\n",
      "\n",
      "== Error ==  system call for: \"['/Users/olgabot/anaconda3/envs/winky/share/spades-3.11.0-0/bin/spades', '/Users/olgabot/code/winky/notebooks/spades-ZN-1/K21/configs/config.info']\" finished abnormally, err code: -6\n",
      "\n",
      "In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru\n",
      "or report an issue on our GitHub repository github.com/ablab/spades\n",
      "Please provide us with params.txt and spades.log files from the output directory.\n"
     ]
    }
   ],
   "source": [
    "! $spades --12 ../winky/test/data/ZN-1-concatenated.fastq -o spades-ZN-1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command line: /Users/olgabot/anaconda3/envs/winky/bin/spades.py\t--plasmid\t--12\t/Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq\t-o\t/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.11.0\n",
      "  Python version: 3.5.2\n",
      "  OS: Darwin-16.7.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: not specified\n",
      "      right reads: not specified\n",
      "      interlaced reads: ['/Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq']\n",
      "      single reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed (with gzip)\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Plasmid mode is turned ON\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/spades.log\n",
      "\n",
      "== Splitting /Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq into left and right reads (in /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input directory)\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /Users/olgabot/anaconda3/envs/winky/share/spades-3.11.0-0/bin/hammer /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  83)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.hpp          :  51)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  93)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  99)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 120)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.001     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.001   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq\n",
      "  0:00:00.008   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.010   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 111)   Total 4 reads processed\n",
      "  0:00:00.010     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.021     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 5214 kmers in total.\n",
      "  0:00:00.021     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.025     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.055     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.056     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 11512 bytes occupied (17.6632 bits per kmer).\n",
      "  0:00:00.057     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 357)   Arranging kmers in hash map order\n",
      "  0:00:00.057     4M / 616M  INFO    General                 (main.cpp                  : 155)   Clustering Hamming graph.\n",
      "  0:00:00.072     4M / 616M  INFO    General                 (main.cpp                  : 162)   Extracting clusters\n",
      "  0:00:00.074     4M / 616M  INFO    General                 (main.cpp                  : 174)   Clustering done. Total clusters: 5194\n",
      "  0:00:00.074     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 379)   Collecting K-mer information, this takes a while.\n",
      "  0:00:00.074     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq\n",
      "  0:00:00.075     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.076     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 392)   Collection done, postprocessing.\n",
      "  0:00:00.076     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 406)   There are 5214 kmers in total. Among them 3374 (64.7104%) are singletons.\n",
      "  0:00:00.076     4M / 616M  INFO    General                 (main.cpp                  : 180)   Subclustering Hamming graph\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 5174. Among them 4208 (81.3297%) are good\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 0. Among them 0 (nan%) are good\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 20. Among them 0 (0%) are good\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2 kmers\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 4208\n",
      "  0:00:00.077     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.971698,0.0188679,0.00943396,0),(0,0.932692,0.0384615,0.0288462),(0.0283019,0.0377358,0.915094,0.0188679),(0,0.00961538,0,0.990385))\n",
      "  0:00:00.077     4M / 616M  INFO    General                 (main.cpp                  : 185)   Finished clustering.\n",
      "  0:00:00.077     4M / 616M  INFO    General                 (main.cpp                  : 204)   Starting solid k-mers expansion in 1 threads.\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (main.cpp                  : 225)   Solid k-mers iteration 0 produced 0 new k-mers.\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (main.cpp                  : 229)   Solid k-mers finalized\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (hammer_tools.cpp          : 211)   Starting read correction in 1 threads.\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (hammer_tools.cpp          : 222)   Correcting pair of reads: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq and /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.087    28M / 616M  INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 0 of 2 reads.\n",
      "  0:00:00.088    28M / 616M  INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:00.088    28M / 616M  INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:00.091     4M / 616M  INFO    General                 (hammer_tools.cpp          : 270)   Correction done. Changed 0 bases in 0 reads.\n",
      "  0:00:00.092     4M / 616M  INFO    General                 (hammer_tools.cpp          : 271)   Failed to correct 0 bases out of 3824.\n",
      "  0:00:00.092     4M / 616M  INFO    General                 (main.cpp                  : 255)   Too few reads have changed in this iteration. Exiting.\n",
      "  0:00:00.092     4M / 616M  INFO    General                 (main.cpp                  : 262)   Saving corrected dataset description to /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/corrected.yaml\n",
      "  0:00:00.092     4M / 616M  INFO    General                 (main.cpp                  : 269)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "== Dataset description file was created: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/plasmid_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.hpp          :  51)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  91)   Assembling dataset (/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  61)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  84)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  85)   Converting paired reads\n",
      "  0:00:00.108    68M / 132M  INFO    General                 (binary_converter.hpp      : 159)   2 reads written\n",
      "  0:00:00.113     4M / 132M  INFO    General                 (read_converter.hpp        :  94)   Converting single reads\n",
      "  0:00:00.346   132M / 260M  INFO    General                 (binary_converter.hpp      : 159)   0 reads written\n",
      "  0:00:00.356     4M / 260M  INFO    General                 (graph_construction.hpp    : 120)   Constructing DeBruijn graph for k=21\n",
      "  0:00:00.356     4M / 260M  INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 1 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.356     4M / 260M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.356     4M / 260M  INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.356     4M / 260M  INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 67108864\n",
      "  0:00:00.358   568M / 1G    INFO    General                 (kmer_splitters.hpp        : 297)   Adding contigs from previous K\n",
      "  0:00:00.358     4M / 1G    INFO    General                 (kmer_splitters.hpp        : 310)   Used 8 reads. Maximum read length 985\n",
      "  0:00:00.358     4M / 1G    INFO    General                 (kmer_splitters.hpp        : 311)   Average read length 956\n",
      "  0:00:00.358     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.359     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2607 kmers in total.\n",
      "  0:00:00.359     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.359     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.359     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.359     4M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.359     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.359     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.367   580M / 1G    INFO    General                 (kmer_splitters.hpp        : 381)   Processed 2607 kmers\n",
      "  0:00:00.367   580M / 1G    INFO    General                 (kmer_splitters.hpp        : 386)   Used 2607 kmers.\n",
      "  0:00:00.367     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.377     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2607 kmers in total.\n",
      "  0:00:00.377     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.381     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.406     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.407     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 10368 bytes occupied (31.8159 bits per kmer).\n",
      "  0:00:00.407     4M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  87)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.408     4M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  91)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.408     4M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:00.410     4M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   59 22-mers were removed by early tip clipper\n",
      "  0:00:00.410     4M / 1G    INFO    General                 (graph_construction.hpp    : 136)   Condensing graph\n",
      "  0:00:00.410     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:00.411     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 4 sequences extracted\n",
      "  0:00:00.411     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:00.412     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.412     4M / 1G    INFO    General                 (graph_construction.hpp    : 141)   Building index with from graph\n",
      "  0:00:00.412     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.412     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.412     4M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.412     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.412     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.415   580M / 1G    INFO    General                 (edge_index_builders.hpp   :  80)   Used 8 sequences.\n",
      "  0:00:00.415     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.422     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2548 kmers in total.\n",
      "  0:00:00.422     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.425     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.449     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\r\n",
      "  0:00:00.450     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 10352 bytes occupied (32.5024 bits per kmer).\r\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (edge_index_builders.hpp   : 105)   Collecting k-mer coverage information from graph, this takes a while.\r\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (edge_index.hpp            :  91)   Index refilled\r\n",
      "  0:00:00.451     4M / 1G    INFO    General                 (graph_construction.hpp    : 173)   Filling coverage index\r\n",
      "  0:00:00.451     4M / 1G    INFO    General                 (edge_index_builders.hpp   : 183)   Collecting k-mer coverage information from reads, this takes a while.\r\n",
      "  0:00:00.451     4M / 1G    INFO    General                 (graph_construction.hpp    : 175)   Filling coverage and flanking coverage from index\r\n",
      "  0:00:00.452     4M / 1G    INFO    General                 (construction.cpp          :  30)   Figured out: read length = 985\r\n",
      "  0:00:00.452     4M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == EC Threshold Finding\r\n",
      "=== Stack Trace ===\r\n",
      "0   spades                              0x0000000106c6ad6d _ZN5utils16print_stacktraceEv + 173\r\n",
      "1   spades                              0x0000000106ed2298 _ZN14coverage_model17KMerCoverageModel3FitEv + 328\r\n",
      "2   spades                              0x0000000106e9948b _ZN14debruijn_graph17GenomicInfoFiller3runERNS_10graph_packINS_13DeBruijnGraphEEEPKc + 1947\r\n",
      "3   spades                              0x0000000106e65ff3 _ZN6spades12StageManager3runERN14debruijn_graph10graph_packINS1_13DeBruijnGraphEEEPKc + 931\r\n",
      "4   spades                              0x0000000106c680ab _ZN6spades15assemble_genomeEv + 7531\r\n",
      "5   spades                              0x0000000106c69bbe main + 2894\r\n",
      "6   libdyld.dylib                       0x00007fff9591c235 start + 1\r\n",
      "Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.\r\n",
      "Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.\r\n",
      "Assertion failed: (cov_.size() > 10), function Fit, file /anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp, line 187.\r\n",
      "\r\n",
      "\r\n",
      "== Error ==  system call for: \"['/Users/olgabot/anaconda3/envs/winky/share/spades-3.11.0-0/bin/spades', '/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/config.info', '/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/plasmid_mode.info']\" finished abnormally, err code: -6\r\n",
      "\r\n",
      "In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru\r\n",
      "or report an issue on our GitHub repository github.com/ablab/spades\r\n",
      "Please provide us with params.txt and spades.log files from the output directory.\r\n"
     ]
    }
   ],
   "source": [
    "! $spades --plasmid --12 ../winky/test/data/ZN-1-concatenated.fastq -o spades-ZN-1-plasmid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command line: /Users/olgabot/anaconda3/envs/winky/bin/spades.py\t--debug\t--plasmid\t--12\t/Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq\t-o\t/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.11.0\n",
      "  Python version: 3.5.2\n",
      "  OS: Darwin-16.7.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned ON\n",
      "\n",
      "Dataset parameters:\n",
      "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: not specified\n",
      "      right reads: not specified\n",
      "      interlaced reads: ['/Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq']\n",
      "      single reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed (with gzip)\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Plasmid mode is turned ON\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/spades.log\n",
      "\n",
      "== Splitting /Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq into left and right reads (in /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input directory)\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /Users/olgabot/anaconda3/envs/winky/share/spades-3.11.0-0/bin/hammer /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  83)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.hpp          :  51)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  93)   Trying to determine PHRED offset\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  99)   Determined value is 33\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  : 120)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.001     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.001   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq\n",
      "  0:00:00.007   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.009   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 111)   Total 4 reads processed\n",
      "  0:00:00.010     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.019     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 5214 kmers in total.\n",
      "  0:00:00.019     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.023     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.053     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.054     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 11512 bytes occupied (17.6632 bits per kmer).\n",
      "  0:00:00.054     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 357)   Arranging kmers in hash map order\n",
      "  0:00:00.055     4M / 616M  INFO    General                 (main.cpp                  : 155)   Clustering Hamming graph.\n",
      "  0:00:00.070     4M / 616M  INFO    General                 (main.cpp                  : 162)   Extracting clusters\n",
      "  0:00:00.072     4M / 616M  INFO    General                 (main.cpp                  : 174)   Clustering done. Total clusters: 5194\n",
      "  0:00:00.072     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 379)   Collecting K-mer information, this takes a while.\n",
      "  0:00:00.072     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq\n",
      "  0:00:00.073     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.074     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 392)   Collection done, postprocessing.\n",
      "  0:00:00.074     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 406)   There are 5214 kmers in total. Among them 3374 (64.7104%) are singletons.\n",
      "  0:00:00.074     4M / 616M  INFO    General                 (main.cpp                  : 180)   Subclustering Hamming graph\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 5174. Among them 4208 (81.3297%) are good\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 0. Among them 0 (nan%) are good\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 20. Among them 0 (0%) are good\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2 kmers\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 4208\n",
      "  0:00:00.076     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.971698,0.0188679,0.00943396,0),(0,0.932692,0.0384615,0.0288462),(0.0283019,0.0377358,0.915094,0.0188679),(0,0.00961538,0,0.990385))\n",
      "  0:00:00.076     4M / 616M  INFO    General                 (main.cpp                  : 185)   Finished clustering.\n",
      "  0:00:00.076     4M / 616M  INFO    General                 (main.cpp                  : 204)   Starting solid k-mers expansion in 1 threads.\n",
      "  0:00:00.076     4M / 616M  INFO    General                 (main.cpp                  : 225)   Solid k-mers iteration 0 produced 0 new k-mers.\n",
      "  0:00:00.076     4M / 616M  INFO    General                 (main.cpp                  : 229)   Solid k-mers finalized\n",
      "  0:00:00.076     4M / 616M  INFO    General                 (hammer_tools.cpp          : 211)   Starting read correction in 1 threads.\n",
      "  0:00:00.076     4M / 616M  INFO    General                 (hammer_tools.cpp          : 222)   Correcting pair of reads: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq and /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.086    28M / 616M  INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 0 of 2 reads.\n",
      "  0:00:00.087    28M / 616M  INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:00.087    28M / 616M  INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:00.090     4M / 616M  INFO    General                 (hammer_tools.cpp          : 270)   Correction done. Changed 0 bases in 0 reads.\n",
      "  0:00:00.090     4M / 616M  INFO    General                 (hammer_tools.cpp          : 271)   Failed to correct 0 bases out of 3824.\n",
      "  0:00:00.090     4M / 616M  INFO    General                 (main.cpp                  : 255)   Too few reads have changed in this iteration. Exiting.\n",
      "  0:00:00.091     4M / 616M  INFO    General                 (main.cpp                  : 262)   Saving corrected dataset description to /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/corrected.yaml\n",
      "  0:00:00.091     4M / 616M  INFO    General                 (main.cpp                  : 269)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "== Dataset description file was created: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/plasmid_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.hpp          :  51)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  91)   Assembling dataset (/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  61)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  79)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  84)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  85)   Converting paired reads\n",
      "  0:00:00.110    68M / 132M  INFO    General                 (binary_converter.hpp      : 159)   2 reads written\n",
      "  0:00:00.116     4M / 132M  INFO    General                 (read_converter.hpp        :  94)   Converting single reads\n",
      "  0:00:00.375   132M / 260M  INFO    General                 (binary_converter.hpp      : 159)   0 reads written\n",
      "  0:00:00.393     4M / 260M  INFO    General                 (graph_construction.hpp    : 120)   Constructing DeBruijn graph for k=21\n",
      "  0:00:00.393     4M / 260M  INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 1 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.393     4M / 260M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.393     4M / 260M  INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.393     4M / 260M  INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 67108864\n",
      "  0:00:00.394   568M / 1G    INFO    General                 (kmer_splitters.hpp        : 297)   Adding contigs from previous K\n",
      "  0:00:00.394     4M / 1G    INFO    General                 (kmer_splitters.hpp        : 310)   Used 8 reads. Maximum read length 985\n",
      "  0:00:00.394     4M / 1G    INFO    General                 (kmer_splitters.hpp        : 311)   Average read length 956\n",
      "  0:00:00.394     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.395     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2607 kmers in total.\n",
      "  0:00:00.395     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.396     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.396     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.396     4M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.396     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.396     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.402   580M / 1G    INFO    General                 (kmer_splitters.hpp        : 381)   Processed 2607 kmers\n",
      "  0:00:00.402   580M / 1G    INFO    General                 (kmer_splitters.hpp        : 386)   Used 2607 kmers.\n",
      "  0:00:00.402     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.414     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2607 kmers in total.\n",
      "  0:00:00.414     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.418     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.444     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.445     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 10368 bytes occupied (31.8159 bits per kmer).\n",
      "  0:00:00.446     4M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  87)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.446     4M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  91)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.446     4M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:00.448     4M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   59 22-mers were removed by early tip clipper\n",
      "  0:00:00.448     4M / 1G    INFO    General                 (graph_construction.hpp    : 136)   Condensing graph\n",
      "  0:00:00.448     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:00.449     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 4 sequences extracted\n",
      "  0:00:00.450     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:00.450     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (graph_construction.hpp    : 141)   Building index with from graph\n",
      "  0:00:00.450     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.454   580M / 1G    INFO    General                 (edge_index_builders.hpp   :  80)   Used 8 sequences.\n",
      "  0:00:00.454     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.462     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2548 kmers in total.\n",
      "  0:00:00.462     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.466     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\r\n",
      "  0:00:00.491     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\r\n",
      "  0:00:00.492     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 10352 bytes occupied (32.5024 bits per kmer).\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (edge_index_builders.hpp   : 105)   Collecting k-mer coverage information from graph, this takes a while.\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (edge_index.hpp            :  91)   Index refilled\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (graph_construction.hpp    : 173)   Filling coverage index\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (edge_index_builders.hpp   : 183)   Collecting k-mer coverage information from reads, this takes a while.\r\n",
      "  0:00:00.494     4M / 1G    INFO    General                 (graph_construction.hpp    : 175)   Filling coverage and flanking coverage from index\r\n",
      "  0:00:00.494     4M / 1G    INFO    General                 (construction.cpp          :  30)   Figured out: read length = 985\r\n",
      "  0:00:00.494     4M / 1G    INFO    General                 (stage.cpp                 :  32)   Saving current state to /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid//K21/saves/construction\r\n",
      "  0:00:00.498     4M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == EC Threshold Finding\r\n",
      "=== Stack Trace ===\r\n",
      "0   spades                              0x0000000102940d6d _ZN5utils16print_stacktraceEv + 173\r\n",
      "1   spades                              0x0000000102ba8298 _ZN14coverage_model17KMerCoverageModel3FitEv + 328\r\n",
      "2   spades                              0x0000000102b6f48b _ZN14debruijn_graph17GenomicInfoFiller3runERNS_10graph_packINS_13DeBruijnGraphEEEPKc + 1947\r\n",
      "3   spades                              0x0000000102b3bff3 _ZN6spades12StageManager3runERN14debruijn_graph10graph_packINS1_13DeBruijnGraphEEEPKc + 931\r\n",
      "4   spades                              0x000000010293e0ab _ZN6spades15assemble_genomeEv + 7531\r\n",
      "5   spades                              0x000000010293fbbe main + 2894\r\n",
      "6   libdyld.dylib                       0x00007fff9591c235 start + 1\r\n",
      "Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.\r\n",
      "Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.\r\n",
      "Assertion failed: (cov_.size() > 10), function Fit, file /anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp, line 187.\r\n",
      "\r\n",
      "\r\n",
      "== Error ==  system call for: \"['/Users/olgabot/anaconda3/envs/winky/share/spades-3.11.0-0/bin/spades', '/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/config.info', '/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/plasmid_mode.info']\" finished abnormally, err code: -6\r\n",
      "\r\n",
      "In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru\r\n",
      "or report an issue on our GitHub repository github.com/ablab/spades\r\n",
      "Please provide us with params.txt and spades.log files from the output directory.\r\n"
     ]
    }
   ],
   "source": [
    "! $spades --debug --plasmid --12 ../winky/test/data/ZN-1-concatenated.fastq -o spades-ZN-1-plasmid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 72\r\n",
      "drwxr-xr-x  12 olgabot  staff   408B Sep 15 15:36 \u001b[1m\u001b[36m.\u001b[m\u001b[m/\r\n",
      "drwxr-xr-x@ 31 olgabot  staff   1.0K Sep 15 15:38 \u001b[1m\u001b[36m..\u001b[m\u001b[m/\r\n",
      "drwxr-xr-x   5 olgabot  staff   170B Sep 15 15:36 \u001b[1m\u001b[36m.bin_reads\u001b[m\u001b[m/\r\n",
      "drwxr-xr-x   3 olgabot  staff   102B Sep 15 15:36 \u001b[1m\u001b[36mK21\u001b[m\u001b[m/\r\n",
      "drwxr-xr-x   7 olgabot  staff   238B Sep 15 15:36 \u001b[1m\u001b[36mconfigs\u001b[m\u001b[m/\r\n",
      "drwxr-xr-x   7 olgabot  staff   238B Sep 15 15:36 \u001b[1m\u001b[36mcorrected\u001b[m\u001b[m/\r\n",
      "-rw-r--r--   1 olgabot  staff    79B Sep 15 15:36 dataset.info\r\n",
      "-rw-r--r--   1 olgabot  staff   262B Sep 15 15:36 input_dataset.yaml\r\n",
      "-rw-r--r--   1 olgabot  staff   1.3K Sep 15 15:36 params.txt\r\n",
      "-rw-r--r--   1 olgabot  staff    21K Sep 15 15:36 spades.log\r\n",
      "drwxr-xr-x   4 olgabot  staff   136B Sep 15 15:36 \u001b[1m\u001b[36msplit_input\u001b[m\u001b[m/\r\n",
      "drwxr-xr-x   2 olgabot  staff    68B Sep 15 15:36 \u001b[1m\u001b[36mtmp\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "ls -lha /Users/olgabot/code/winky/notebooks/spades-ZN-1/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 16\r\n",
      "drwxr-xr-x   4 olgabot  staff   136B Sep 15 15:36 \u001b[1m\u001b[36m.\u001b[m\u001b[m/\r\n",
      "drwxr-xr-x  12 olgabot  staff   408B Sep 15 15:36 \u001b[1m\u001b[36m..\u001b[m\u001b[m/\r\n",
      "-rw-r--r--   1 olgabot  staff   3.7K Sep 15 15:36 ZN-1-concatenated_1.fastq\r\n",
      "-rw-r--r--   1 olgabot  staff   3.8K Sep 15 15:36 ZN-1-concatenated_2.fastq\r\n"
     ]
    }
   ],
   "source": [
    "ls -lha /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       8 /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_1.fastq\r\n",
      "       8 /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_2.fastq\r\n",
      "      16 total\r\n"
     ]
    }
   ],
   "source": [
    "! wc -l /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_1.fastq <==\r\n",
      "@ZN-1_ZNif1_\r\n",
      "ATGCGAAAAACTTTCTGATCAGTTCCTGTTGATTCCCAAGGTATGCATATGGTAGCCGGGCACGACGCCAATGATGCTGTCATAAGCAATAGCGTTGCGCAAGCGCGATTTTCAGGGCTTCTGATCGTGAAAACAGTTCTCGATCATATTCTGCAAAAGACAGAACGCGGCGTCCGATTGCATCCTCTCGCACGGACGGCTAAGGTGAAAAATGAGGTCAACTCTTTCAAGGCCGCACTGAGCTCTCTGGCAAAACACGGGGAATATGCCCCATTCGCGAGACTGCTTAATTTGTCTGGAGTCAACAATCTGGAGCATGGGCTTTTTCCGCAACTTAGTGCGATTGCCCTGGGGGTAGCTACAGCACACGGTAGCACTTTGGCCGGCGTTAACGTCGGTGAGCAATATCAACAACTGCGAGAAGCAGCGACGGAGGCAGAGAAACAACTGCAACAGTATGCCGAGTCTAGGGAACTGGATCACCTCGGTCTGGATGACCAGGAAAAGAAAATTCTCATGAACTTTCATCAAAAGAAGAACGAGATATCATTTCAGCAGACTAATGCGATGGTCACTCTTCGGAAAGAGCGACTTGCTAAACTCACAGAAGCGATAACAGCAGCATCCTTGCCGAAGACATCAGGCCACTATGACGACGACGACGATATTCCCTTTCCGGGTCCGATTAACGATGACGATAACCCGGGTCATCAAGACGATGATCCCACAGACTCCCAGGACACCACCATCCCCGATGTAGTTGTGGATCCTGATGATGGCTCATACGGCGAGTACCAAAGCTATAGTGAGAATGGCATGAACGCCCCGGGACGATTTGGTATTGTTCGACCTCGATGAAGATGACGAGGATACTAAGCCTGTACCCAACCGAAGTACCAAAGGGCG\r\n",
      "+\r\n",
      "/5E:.1'&&&1VVK/BP)',&,J^PPG^PX^/--+,**4GPP^R^RXX^^^^X^^^P^^^P^^^\\\\^^R-\\^\\L\\^^^^\\>L\\X^\\R\\^^^^^^^\\\\^^^^\\R^^^^\\?\\^^3L^^\\^^^^^^RR\\^^^^^^^^\\^^^^^^^\\MLD^^^\\\\^^^^^^^RR^^\\\\^^\\^^^^^^^^^\\\\^^^^^^^^^^^^^^^^\\M\\^^^\\\\^^^^^^^^^^^^^^^^\\\\\\\\^^^^^\\\\^^^^^^^^\\^^\\\\^\\^^^^^^^^^^^^^^^^^^\\^^^^\\^^^^^^^^^^^\\RL\\^^^^^^\\^^^^^^^^^\\\\^\\\\^^^^^^^^^RR^^^^^XX^XX^^^^^^^^R^^^^^^^^^^^^^^^^^^^^^^^^^RR^^^^^^^^^^^^^^^^^X^^^^^^^^^^^^^^^^^^^^^^^^^^^^^PP^^^^^^^^^^PP^^^^^^^^^^^>A^^^^LGL^P^^^^^^^^V^^^^^^^^^^^P^^^^^^^^^K^^^KHGI^^^^^^^^^ZZHKZZKE?ZZPFPZKOZZVUZZZEVZZZCPZZZ?LTEIZZP9PZPVAKVZVENVZEJZZZ;FZVZVEAEBPVZVUVSZUJBGOLOZZUZZJLUIE<ZGVIZZVGLAZUIZAUVVPJOCZPUUBPZPZVVSVUZHUZZU<ZUVU2>GDZ@UNZG8AIUFVVIG4UV5VM0NQ9VV<LMEZHZZO@ZBOVNMZBD0=DJG?P??HC:MQ@MAC<OTMP:CI??N?>::OEHI:LO<>E>@48QCT?E0?<>>>>>@:>G;TTPP;5OBE?<EJ4LL=>=>>O98?DA>C9O5D>>A8L18J;;8B31J6;>><L>=>L>=>>8DO;>DG;8H>>TOH-E0562>4TA>;5=JJE40>=:+=D08558E>371;==-=7G==5A74G>C@=A1.8K>524?=9::64>B'G+BT6?8\r\n",
      "@ZN-1_pCAGGSf_\r\n",
      "GCTGTCCGCGGGGGGACGGCTGCCTTCGGGGGGGACGGGGCAGGGCGGGGTTCGGCTTCTGGCGTGTGACCGGCGGCTCTAGAGCCTCTGCTAACCATGTTCATGCCTTCTTCTTTTTCCTACAGCTCCTGGGCAACGTGCTGGTTATTGTGCTGTCTCATCATTTTGGCAAAGAATTCATGGATAGCCGCCCACAGAAAATTTGGATGGCACCATCCCTGACCGAAAGTGATATGGACTATCATAAGATTTTGACTGCCGGCTTGAGTGTACAACAAGGGATTGTAAGACAGAGAGTTATCCCGGTCTACCAAGTGAACAATTTGGAAGAGATCTGTCAGCTGATTATCCAGGCGTTTGAAGCGGGCGTAGATTTCCAAGAATCAGCCGATTCTTTTCTCCTTATGCTGTGCCTTCATCACGCATATCAAGGTGATTATAAACTCTTCTTGGAATCTGGCGCGGTCAAATACTTGGAAGGGCATGGTTTCAGATTCGAGGTGAAGAAAAGGGACGGAGTCAAGAGGCTGGAGGAGCTTCTGCCGGCCGTTTCTAGTGGCAAAAACATTAAACGCACACTCGCCGCGATGCCTGAAGAGGAGACTACAGAAGCGAATGCCGGCCAATTTCTGTCATTCGCGTCTTTGTTCCTGCCAAAGCTCGTAGTAGGTGAAAAAGCATGCCTGGAAAAAGTACAGCGCCAAATTCAGGTGCATGCCGAACAAGGTCTTATCCAGTACCCAACGGCTTGGCAGTCAGTTGGCCACATGATGGTGATCTTTAGACTTATGCGAACAAACTTTCTGATCAAGTTCCTGTTGATTCACCAAGGTATGCATATGGTAGCCGGGCACGACGCCAATGATGCTGTCATAAGCAATAGCGTTGCGCAAGCGCGATTTTCAGGGCTTCTGATCGTGAAAACAGTTCTCGATCATATTCTGCAAAAGACAGAACGCGGCGTCCGATTGCATCCTCTCGCACG\r\n",
      "+\r\n",
      ":ZJE.TZ7)J^^^^^+)P^PPP>P^^F^^^^^^^83F^^^\\\\^^\\^\\^^^^^^^^\\^^^^^^^^^^^^\\R^^^^^^^^^^^\\L<M\\^^^^^\\\\^\\?1X^^^<=^^^^^^^^^\\^^^^^^^^^^^X^^\\\\^^^^^^^^^^^^^^^^\\\\^^^^^^^^^^^\\\\^\\\\^^^^^^^^^^\\\\^^^\\\\^^^\\^^^^^^\\^^^^^\\\\^^^^^^^^\\\\^^^^^\\^\\MM\\^^\\\\^^^^\\\\^^^^^^^^^\\R\\^\\\\^^^^\\^^^^^^\\^^^\\^^^^^^^^^^^^^^^^^^^^^\\^^^^^^^\\^^^\\\\\\^^^^^^^^^^^^^^\\\\\\^^^^^^^^^^^^^^^^^^^\\\\^^^^\\\\^^^^^^\\^^^\\\\^^^^^^^^^^^^\\X^^^^^^\\^^^^\\\\^^\\^^\\\\X\\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\^^^^^^^^\\\\X\\^^\\^^\\\\^^^^^^^^\\^^^^^^^^^^^^^\\^^^^^^^^^^^^^^^^^^^^^^^^^^^^XR^^^XX^^^^^^^^^^^^^^^^^P^^^^^^P^^^^^^^^^^^^^^^^^^^^PXX^P^^^^^^^^^^^^^^^^^^^^^P^^^^^^^^^^^^PL^^L^^^^^^^^^^^^P^^^LL?K^^^^^^^I^^^^^^^UHGH^^K^^ZKGIPVZZIHKVZZZZVGZZZZKHDCZZVZZZPSZUUZZZVVZZZZZVZVZVPZZZVZZZZVVUNZJJJFFD1ZJFZJEZZSUUNZJFZUU9USVVZVVZUVZO98VNHLZDD9FUZFZEZ=KUVHCBKKVZ:;83VSVZUEZO?HSSNZZUI:B@Z=IQFI9V5U6Z@UVZVCUIF=:1FPJVBPIZ5PL@9V8AFABI@Z=CJFJCIFMC?<Q:?OQT@Q2B9C97MM&N@QC>ACQG09>Q=O9@.?FD=E?.Q>4.>:@>E6@3>0TTP4>>O>>GH=>>95?>.>?TO3J3E-F6H2;=:=5J=?F2>>2>TT?<297<=<D==9B:=/:/8=067/4;A+;3E155-:6\r\n",
      "\r\n",
      "==> /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/ZN-1-concatenated_2.fastq <==\r\n",
      "@ZN-1_ZNir1_\r\n",
      "GTGCGGTGGGGACCGGGAATTTTGTATCGGTCTGGATTGCGTCTGCCTCCCTTCAATATGTTGTCCCTTTTGGCTGTTTTTTTGCTGGCCGCCCTTGGTACTTCGGTTGGGTACAGGCTTAGTATCCTCGTCATCTTCATCGAGGTCGAACAATACCAAATCGTCCGGGGCGTTCATGCCATTCTCACTATAGCTTTGGTACTCGCCGTATGAGCCATCATCAGGATCCACAACTACATCGGGGATGGTGGTGTCCTGGGAGTCTGTGGGATCATCGTCTTGATGACCCGGGTTATCGTCATCGTTAATCGGACCCGGAAAGGGAATATCGTCGTCGTCGTCATAGTGGCCTGATGTCTTCGGCAAGGATGCTGCTGTTATCGCTTCTGTGAGTTTAGCAAGTCGCTCTTTCCGAAGAGTGACCATCGCATTAGTCTGCTGAAATGATATCTCGTTCTTCTTTTGATGAAAGTTCATGAGAATTTTCTTTTCCTGGTCATCCAGACCGAGGTGATCCAGTTCCCTAGACTCGGCATACTGTTGCAGTTGTTTCTCTGCCTCCGTCGCTGCTTCTCGCAGTTGTTGATATTGCTCACCGACGTTAACGCCGGCCAAAGTGCTACCGTGTGCTGTAGCTACCCCCAGGGCAATCGCACTAAGTTGCGGAAAAAGCCCATGCTCCAGATTGTTGACTCCAGACAAATTAAGCAGTCTCGCGAATGGGGCATATTCCCCGTGTTTTGCCAGAGAGCTCAGTGCGGCCTTGAAAGAGTTGACCTCATTTTTCACCTTAGCCGTCCGTGCGAGAGGATGCAATCGGACGCCGCGTTCTGTCTTTTGCAGAATATGATCGAGAACTGTTTTCACGATCAGAAGCCCTGAAAATCGCGCTTGCGCAACGCTATTGCTTATGACAGCATCATTGGCGTCGTGCCCGGCTACCATATGCAT\r\n",
      "+\r\n",
      "2;UFCJ+JL)()')VJ%%'.,VV^S/8L^VE;F^@+)1XSXPP^^^RLUXXRX,>GR^^^X^^\\\\^\\^\\^^^\\J\\^L^^^^^^^\\^^\\\\\\^N\\^\\\\^\\\\\\^^\\?\\\\N\\\\\\\\UR^L\\^\\\\\\^R\\\\MF^\\\\\\LI\\\\\\\\^A\\\\\\\\\\\\^RUE\\U\\\\U\\\\\\M\\X\\\\\\\\HU^^\\\\\\\\\\UU\\\\^^\\\\L^\\\\\\HR\\X\\\\\\\\\\XX\\\\\\UU^^\\\\\\^^R\\^^^\\^L\\^\\\\^\\\\\\^M\\X\\^^\\U\\\\\\\\U\\R\\\\R\\\\JX\\?\\\\H^R\\^\\\\U\\\\\\\\\\^^U^\\\\\\\\\\\\^\\^M\\^\\^\\\\\\\\^^\\^^\\\\\\\\^\\^L\\^^^^\\\\\\\\^U\\U\\\\^\\\\\\CM\\^^\\\\\\^^\\\\^U\\^\\^^\\^\\R\\\\^\\REX\\\\^^^\\\\^\\\\^^\\^^\\\\U\\^?\\\\\\\\\\R^^\\^\\^\\\\J^\\U^^U^^\\\\\\\\^L^^X^XUUR^U^^^X^X^XIFPR^XRRE^UXUX^^^UUXX^J^^^^X^^^^SXSXRXJXS^^X^XXX^^PX^XSXSXLP^^^EPS^SSSS^^^SSS^^NNGG^S>PP9=^^LI^CS^^^CG^^^SVP^V93C^S^^PVZVP:PEPVVVVOPGGZVVZVZZPVZVVVUSUPZVLZZPZVUVVP;BPFPZHSVZPZPZ;HJZZ<FVULHLJUUUZVC=BZD@DVULCJZP8GPVZZUDV?ZD<UJZZFS@JJC<=ZVIF@LCF:COVVCZVJDGJFFUZ@FJPVZH<DA71GQ:IM?>ZZFBB>D@ZLFGHQDF<Q@IOCJ>@N8<LOK?G?J@MF@TN<MAP<NNLAF>5=@0:=MQ:C@>GEM?*>>A>ETC>B4JE>97>>5=>OLOH5;8<@>?(L>=0OL>-8?2E*68L<:CG:G>>78/;=J?B8>6A->L6<TOT>C=C@A6.3>C@0A=D/:670?:OGO1652>660===7GT-043GG7:=B70?.:++)?28=A7C124A0H8<,.5C<<<692=0++;+2+19(3/M8+.-5.677/039B8A\r\n",
      "@ZN-1_pCAGGSr_\r\n",
      "CACACACTATTGCAATGAAAATAAATTTCCTTTATTAGCCGAAGTCAGATGCTCAAGGGGCTTCATGATGTCCCCATAATTTTTGGCAGAGGGAAAAAGATCTGCTAGCTCGAGCATGCCCGGGTACCATGCATCGATGAGCTCGAATTCTCATTGGTGGTGTTGCAGTATCGCCATAAATTTATTCTTATGATTCATTACAGGCCAGTAGAATTGCTGCCCATCGAGCGTAACAAACCTGTTTTCCTCATTCATCGCCTCCTTTTCCGTAAGCCATGGTGGGTACTCCTCCTCGAGGCTGTCTGGGTATGTATACTCCTTTCCGTCAGAGGTTGAAAAGACGACCGGTTCGTCCTTCATCATGTGATAGTACAATACCGCATCGAACGGCCCTTGGCTCCTCAATATATGCCTGTACATCTCCTCGAATGAGTGCTCAGATTGGGTGTTGTCACTGTCTTGATTTCTAGCCTCTTGCGTATGGTCTTGGTCCTGTTGCTCGTCCTGTGGGAGCTCCTTCTTTTCGGAGTGGTCGCGGTAAACCGGGGCCGGAGGCGCGACGGTAGGAGTCCGGTTAGATGTGCCATCCCTATCTTGCTCTTCGTCGTCAGACTCAAGCGGCGGAAGGCTACTAGTCTCGTCGTCCGCGTCATCCAGCGGATCTGCCTCTTCGTTAATGGGGGTGAGCATCCTAGGACTGGTGGAACCTGAAGGTTCATTTCGTCTATCATTGTCAGTGAGTGGAGCGGAAGCATGGTGAATGGTGCGGTGGGGACCCGGGACATTTTGTATCGGTCTGGATTGCGTCTGCCTCCCTTCAATATGTTGTCCCTTTTGGCTGTTTTTTTGCTGGCCGCCCTTGGTACTTCGGTTGGGTACAGGCTTAGTATCCTCGTCATCTTCATCGAGGTCGAACAATACCAAATCGTCCGGGGCGTTCATGCCATTCTCACTATAGCTTTGGTACTCGCCGTATGAGCCA\r\n",
      "+\r\n",
      "3),+D5K;FKKCG8P^^^^^^A>^P^^^^PR^PURGR049(1^\\^4,^RL^^^FL^^^^^^^^GR^^^\\^^^^^\\^^^^^^^^^^^^\\^\\\\\\\\^^^\\^\\MM^^\\\\\\\\^^^^^^\\\\^^^\\\\\\^^\\^^^\\R^^^^^^^^^\\^^^^^^^^^\\XX\\\\^^^^^^^\\^^^\\^^^^^^^\\\\\\^^^^^^^^^^^^^^^^^^^^\\\\^\\^^\\\\\\^R\\^^^\\^\\\\^^\\^\\\\^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\^^^\\\\^^^^^^^\\^^^^^^^^^XX^\\\\^^^^^^^^^^^\\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\\\^\\M\\^^^^^^\\\\^\\\\\\^^\\\\^^^X^\\^^^^\\^^\\^^^^^^^^^\\^^^^\\\\^^\\^^^\\FX^^^^^^\\^^^^^^XX^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\\\\\\\^^^^^^^^^XR^^^^^^^^^^^^^^^^X^^^^^^^RR^^^^^^^^^^^^^^^^^^^P^^^^^^^^X^^^^^^^^^^^^^^^^^^^^^^^^X^EG^^^^^^^^^^^^^^^^^^^PFP^^L^^^^^^^^^^^^^P^^^^^^PKK?P^^^^^^^^^^^P^^^^^VL^^IPPZIKZPPZZZZV<ZPKOZVPKBP@KVZZZZKIZKAZZH9PVNUZZIHZVZZJPZZE9UZUZZVHBHBSUZOPPJAJQUVUZUZUZ8ZD@ZZ@9ZJZ?CZZZPBCU?V<@ZZVVFKVJFG8DUZT9@;UHUSSHZPV@FJEVSEVEPZZAKSFFZU9?@OV3Z8.89:JJLZC2BBFZZUV=EUPUV?VMZIBBDA8C=IIA-DV?ZCML8>@FIJV9IL-PZIVTT@QF=AL-TOOTTO>5H@5>Q@,O?G62?38L?EDHQ;FLT?50Q2EH>M=>:8@B<>>9>45:>=@E48>>;C@L5L/B=B/:735()OA=D<+A=BTO1C@7>79=;.,1<A.?0:8=0=0@=ET@=@/2<>6<;7;)8;>:-?97\r\n"
     ]
    }
   ],
   "source": [
    "! head /Users/olgabot/code/winky/notebooks/spades-ZN-1/split_input/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ZN-1-concatenated.fastq             \u001b[31mZN-2_ZNif1__2017-06-07_C01.ab1\u001b[m\u001b[m*\r\n",
      "\u001b[31mZN-1_ZNif1__2017-06-07_B01.ab1\u001b[m\u001b[m*     \u001b[31mZN-2_ZNir1__2017-06-07_F01.ab1\u001b[m\u001b[m*\r\n",
      "ZN-1_ZNif1__2017-06-07_B01.fastq    \u001b[31mZN-2_pCAGGSf__2017-05-31_B06.ab1\u001b[m\u001b[m*\r\n",
      "\u001b[31mZN-1_ZNir1__2017-06-07_E01.ab1\u001b[m\u001b[m*     \u001b[31mZN-2_pCAGGSr__2017-05-31_H03.ab1\u001b[m\u001b[m*\r\n",
      "ZN-1_ZNir1__2017-06-07_E01.fastq    \u001b[31mZN-3_ZNif1__2017-06-07_D01.ab1\u001b[m\u001b[m*\r\n",
      "\u001b[31mZN-1_pCAGGSf__2017-05-31_A06.ab1\u001b[m\u001b[m*   \u001b[31mZN-3_ZNir1__2017-06-07_G01.ab1\u001b[m\u001b[m*\r\n",
      "ZN-1_pCAGGSf__2017-05-31_A06.fastq  \u001b[31mZN-3_pCAGGSf__2017-05-31_C06.ab1\u001b[m\u001b[m*\r\n",
      "\u001b[31mZN-1_pCAGGSr__2017-05-31_G03.ab1\u001b[m\u001b[m*   \u001b[31mZN-3_pCAGGSr__2017-05-31_A04.ab1\u001b[m\u001b[m*\r\n",
      "ZN-1_pCAGGSr__2017-05-31_G03.fastq\r\n"
     ]
    }
   ],
   "source": [
    "ls ../winky/test/data/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command line: /Users/olgabot/anaconda3/envs/winky/bin/spades.py\t--cov-cutoff\toff\t--debug\t--plasmid\t--12\t/Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq\t-o\t/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.11.0\n",
      "  Python version: 3.5.2\n",
      "  OS: Darwin-16.7.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned ON\n",
      "\n",
      "Dataset parameters:\n",
      "  Multi-cell mode (you should set '--sc' flag if input data was obtained with MDA (single-cell) technology or --meta flag if processing metagenomic dataset)\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: not specified\n",
      "      right reads: not specified\n",
      "      interlaced reads: ['/Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq']\n",
      "      single reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed (with gzip)\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Plasmid mode is turned ON\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/spades.log\n",
      "\n",
      "== Splitting /Users/olgabot/code/winky/winky/test/data/ZN-1-concatenated.fastq into left and right reads (in /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input directory)\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /Users/olgabot/anaconda3/envs/winky/share/spades-3.11.0-0/bin/hammer /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  83)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  84)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.hpp          :  51)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  93)   Trying to determine PHRED offset\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  99)   Determined value is 33\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  : 120)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.001     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.001   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq\n",
      "  0:00:00.008   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             :  96)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.011   580M / 616M  INFO   K-mer Splitting          (kmer_data.cpp             : 111)   Total 4 reads processed\n",
      "  0:00:00.011     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.021     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 5214 kmers in total.\n",
      "  0:00:00.021     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.025     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.055     4M / 616M  INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.056     4M / 616M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 11512 bytes occupied (17.6632 bits per kmer).\n",
      "  0:00:00.056     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 357)   Arranging kmers in hash map order\n",
      "  0:00:00.057     4M / 616M  INFO    General                 (main.cpp                  : 155)   Clustering Hamming graph.\n",
      "  0:00:00.073     4M / 616M  INFO    General                 (main.cpp                  : 162)   Extracting clusters\n",
      "  0:00:00.075     4M / 616M  INFO    General                 (main.cpp                  : 174)   Clustering done. Total clusters: 5194\n",
      "  0:00:00.075     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 379)   Collecting K-mer information, this takes a while.\n",
      "  0:00:00.075     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq\n",
      "  0:00:00.076     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 385)   Processing /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.077     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 392)   Collection done, postprocessing.\n",
      "  0:00:00.077     4M / 616M  INFO   K-mer Counting           (kmer_data.cpp             : 406)   There are 5214 kmers in total. Among them 3374 (64.7104%) are singletons.\n",
      "  0:00:00.077     4M / 616M  INFO    General                 (main.cpp                  : 180)   Subclustering Hamming graph\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 5174. Among them 4208 (81.3297%) are good\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 0. Among them 0 (nan%) are good\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 20. Among them 0 (0%) are good\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2 kmers\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 4208\n",
      "  0:00:00.078     4M / 616M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.971698,0.0188679,0.00943396,0),(0,0.932692,0.0384615,0.0288462),(0.0283019,0.0377358,0.915094,0.0188679),(0,0.00961538,0,0.990385))\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (main.cpp                  : 185)   Finished clustering.\n",
      "  0:00:00.078     4M / 616M  INFO    General                 (main.cpp                  : 204)   Starting solid k-mers expansion in 1 threads.\n",
      "  0:00:00.079     4M / 616M  INFO    General                 (main.cpp                  : 225)   Solid k-mers iteration 0 produced 0 new k-mers.\n",
      "  0:00:00.079     4M / 616M  INFO    General                 (main.cpp                  : 229)   Solid k-mers finalized\n",
      "  0:00:00.079     4M / 616M  INFO    General                 (hammer_tools.cpp          : 211)   Starting read correction in 1 threads.\n",
      "  0:00:00.079     4M / 616M  INFO    General                 (hammer_tools.cpp          : 222)   Correcting pair of reads: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_1.fastq and /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/split_input/ZN-1-concatenated_2.fastq\n",
      "  0:00:00.088    28M / 616M  INFO    General                 (hammer_tools.cpp          : 166)   Prepared batch 0 of 2 reads.\n",
      "  0:00:00.089    28M / 616M  INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:00.089    28M / 616M  INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:00.093     4M / 616M  INFO    General                 (hammer_tools.cpp          : 270)   Correction done. Changed 0 bases in 0 reads.\n",
      "  0:00:00.093     4M / 616M  INFO    General                 (hammer_tools.cpp          : 271)   Failed to correct 0 bases out of 3824.\n",
      "  0:00:00.093     4M / 616M  INFO    General                 (main.cpp                  : 255)   Too few reads have changed in this iteration. Exiting.\n",
      "  0:00:00.093     4M / 616M  INFO    General                 (main.cpp                  : 262)   Saving corrected dataset description to /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/corrected.yaml\n",
      "  0:00:00.093     4M / 616M  INFO    General                 (main.cpp                  : 269)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "== Dataset description file was created: /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/plasmid_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.hpp          :  51)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  91)   Assembling dataset (/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  61)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  79)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 126)   STAGE == Construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  84)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  85)   Converting paired reads\n",
      "  0:00:00.111    68M / 132M  INFO    General                 (binary_converter.hpp      : 159)   2 reads written\n",
      "  0:00:00.117     4M / 132M  INFO    General                 (read_converter.hpp        :  94)   Converting single reads\n",
      "  0:00:00.385   132M / 260M  INFO    General                 (binary_converter.hpp      : 159)   0 reads written\n",
      "  0:00:00.397     4M / 260M  INFO    General                 (graph_construction.hpp    : 120)   Constructing DeBruijn graph for k=21\n",
      "  0:00:00.398     4M / 260M  INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 1 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.398     4M / 260M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.398     4M / 260M  INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.398     4M / 260M  INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 67108864\n",
      "  0:00:00.399   568M / 1G    INFO    General                 (kmer_splitters.hpp        : 297)   Adding contigs from previous K\n",
      "  0:00:00.399     4M / 1G    INFO    General                 (kmer_splitters.hpp        : 310)   Used 8 reads. Maximum read length 985\n",
      "  0:00:00.399     4M / 1G    INFO    General                 (kmer_splitters.hpp        : 311)   Average read length 956\n",
      "  0:00:00.399     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.400     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2607 kmers in total.\n",
      "  0:00:00.400     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.401     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.401     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.401     4M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.401     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.401     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.405   580M / 1G    INFO    General                 (kmer_splitters.hpp        : 381)   Processed 2607 kmers\n",
      "  0:00:00.405   580M / 1G    INFO    General                 (kmer_splitters.hpp        : 386)   Used 2607 kmers.\n",
      "  0:00:00.406     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.414     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2607 kmers in total.\n",
      "  0:00:00.414     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.417     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n",
      "  0:00:00.443     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\n",
      "  0:00:00.445     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 10368 bytes occupied (31.8159 bits per kmer).\n",
      "  0:00:00.445     4M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  87)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.446     4M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  91)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.446     4M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:00.448     4M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   59 22-mers were removed by early tip clipper\n",
      "  0:00:00.448     4M / 1G    INFO    General                 (graph_construction.hpp    : 136)   Condensing graph\n",
      "  0:00:00.448     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:00.449     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 4 sequences extracted\n",
      "  0:00:00.450     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:00.450     4M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (graph_construction.hpp    : 141)   Building index with from graph\n",
      "  0:00:00.450     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 289)   Building kmer index\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 106)   Splitting kmer instances into 16 buckets using 1 threads. This might take a while.\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 256\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  74)   Memory available for splitting buffers: 83.332 Gb\n",
      "  0:00:00.450     4M / 1G    INFO    General                 (kmer_splitters.hpp        :  82)   Using cell size of 4194304\n",
      "  0:00:00.454   580M / 1G    INFO    General                 (edge_index_builders.hpp   :  80)   Used 8 sequences.\n",
      "  0:00:00.455     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 109)   Starting k-mer counting.\n",
      "  0:00:00.462     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 115)   K-mer counting done. There are 2548 kmers in total.\n",
      "  0:00:00.463     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 121)   Merging temporary buckets.\n",
      "  0:00:00.465     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 298)   Building perfect hash indices\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.490     4M / 1G    INFO    General                 (kmer_index_builder.hpp    : 137)   Merging final buckets.\r\n",
      "  0:00:00.492     4M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 320)   Index built. Total 10352 bytes occupied (32.5024 bits per kmer).\r\n",
      "  0:00:00.492     4M / 1G    INFO    General                 (edge_index_builders.hpp   : 105)   Collecting k-mer coverage information from graph, this takes a while.\r\n",
      "  0:00:00.492     4M / 1G    INFO    General                 (edge_index.hpp            :  91)   Index refilled\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (graph_construction.hpp    : 173)   Filling coverage index\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (edge_index_builders.hpp   : 183)   Collecting k-mer coverage information from reads, this takes a while.\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (graph_construction.hpp    : 175)   Filling coverage and flanking coverage from index\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (construction.cpp          :  30)   Figured out: read length = 985\r\n",
      "  0:00:00.493     4M / 1G    INFO    General                 (stage.cpp                 :  32)   Saving current state to /Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid//K21/saves/construction\r\n",
      "  0:00:00.495     4M / 1G    INFO   StageManager             (stage.cpp                 : 126)   STAGE == EC Threshold Finding\r\n",
      "=== Stack Trace ===\r\n",
      "0   spades                              0x00000001004e2d6d _ZN5utils16print_stacktraceEv + 173\r\n",
      "1   spades                              0x000000010074a298 _ZN14coverage_model17KMerCoverageModel3FitEv + 328\r\n",
      "2   spades                              0x000000010071148b _ZN14debruijn_graph17GenomicInfoFiller3runERNS_10graph_packINS_13DeBruijnGraphEEEPKc + 1947\r\n",
      "3   spades                              0x00000001006ddff3 _ZN6spades12StageManager3runERN14debruijn_graph10graph_packINS1_13DeBruijnGraphEEEPKc + 931\r\n",
      "4   spades                              0x00000001004e00ab _ZN6spades15assemble_genomeEv + 7531\r\n",
      "5   spades                              0x00000001004e1bbe main + 2894\r\n",
      "6   libdyld.dylib                       0x00007fff9591c235 start + 1\r\n",
      "Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.\r\n",
      "Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.\r\n",
      "Assertion failed: (cov_.size() > 10), function Fit, file /anaconda/conda-bld/spades_1505345597413/work/SPAdes-3.11.0/src/common/modules/coverage_model/kmer_coverage_model.cpp, line 187.\r\n",
      "\r\n",
      "\r\n",
      "== Error ==  system call for: \"['/Users/olgabot/anaconda3/envs/winky/share/spades-3.11.0-0/bin/spades', '/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/config.info', '/Users/olgabot/code/winky/notebooks/spades-ZN-1-plasmid/K21/configs/plasmid_mode.info']\" finished abnormally, err code: -6\r\n",
      "\r\n",
      "In case you have troubles running SPAdes, you can write to spades.support@cab.spbu.ru\r\n",
      "or report an issue on our GitHub repository github.com/ablab/spades\r\n",
      "Please provide us with params.txt and spades.log files from the output directory.\r\n"
     ]
    }
   ],
   "source": [
    "! $spades --cov-cutoff off --debug --plasmid --12 ../winky/test/data/ZN-1-concatenated.fastq -o spades-ZN-1-plasmid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
